frontend/controller/actions/identity-kv.js

```diff
@@ -176,8 +186,92 @@ export default (sbp('sbp/selectors/register', {
         return { ...currentPreferences, hideDistributionBanner }
       }
 
-      const data = await getUpdatedPreferences(contractID)
+      const data = await getUpdatedPreferences()
       await sbp('gi.actions/identity/kv/savePreferences', { data, onconflict: getUpdatedPreferences })
     })
+  },
+  // Notifications
+  'gi.actions/identity/kv/fetchNotificationStatus': async () => {
+    const { ourIdentityContractId } = sbp('state/vuex/getters')
+    if (!ourIdentityContractId) {
+      throw new Error('Unable to fetch notification status without an active session')
+    }
+    return (await sbp('chelonia/kv/get', ourIdentityContractId, KV_KEYS.NOTIFICATIONS))?.data || {}
+  },
+  'gi.actions/identity/kv/saveNotificationStatus': ({ data, onconflict }: { data: Object, onconflict?: Function }) => {
+    const { ourIdentityContractId } = sbp('state/vuex/getters')
+    if (!ourIdentityContractId) {
+      throw new Error('Unable to update notification status without an active session')
+    }
+
+    // const enforceStorageRules = (notificationStatus) => {
+    //   return sbp('state/vuex/getters').notifications
+    //     .map(notification => notification.hash)
+    //     .reduce((acc, hash) => ({ ...acc, [hash]: notificationStatus[hash] }), {})
+    // }
+
+    // const updatedOnConflict = async (...args) => {
+    //   if (typeof onconflict === 'function') {
+    //     return enforceStorageRules(await onconflict(...args))
+    //   }
+    //   return null
+    // }
```

Comment:
To comment these out was my intentional to keep in mind that we should save only notifications which are applied the storage rules. (by `applyStorageRules` function)

But, I don't think we should do this in here, where we save notification status in KV store, with the following reasons.
- Each device could have different notifications, and storage-rule-applied notifications could differ from one to another.
- I don't think to save all of them will cause a memory problem since they are very simple object.

@taoeffect, if you still want to apply storage rules in saving notification status in KV store, I can restore this code.
And if you agree with me, I will remove them and add this comment instead.

Reply:
I think we should figure out a way to have the best of all worlds.

In other words, I think you are correct that we need to be able to allow different machines to have different notifications, while at the same time we should try to avoid keeping a record of all notification statuses forever in the kv-store. Even though you are right that they are a small amount of data, over a long enough period of time this could potentially cause problems.

So let's apply different storage rules to statuses stored in the KV store.

`initNotificationStatus`, instead of being an object, can be a function that returns an object, and that will allow us to store a `createdDate` field on it.

We can use that field to remove all notification statuses that are older than the oldest notifications possible that are stored in the Vuex state.

Based on the rules of `applyStorageRules` we can apply the following rule here:

- Delete all read notifications that are older than `MAX_AGE_READ` (see `storageConstants.js`)

This will allow us to safely store all notifications from all devices on the server while keeping their number reasonably limited.

We should also add a comment here in the code to explain this reasoning.

Reply:
👍